traefik/templates/hpa.yaml

```
@@ -1,5 +1,5 @@
 {{- if .Values.autoscaling.enabled }}
-apiVersion: autoscaling/v2beta1
+apiVersion: autoscaling/v2beta2
```

```suggestion
{{- if .Capabilities.APIVersions.Has "autoscaling/v2beta2/HorizontalPodAutoscaler" }}
apiVersion: autoscaling/v2beta2
{{- else if .Capabilities.APIVersions.Has "autoscaling/v2beta1/HorizontalPodAutoscaler" }}
apiVersion: autoscaling/v2beta1
{{- else }}
{{- fail "\n\n ERROR: You must have at least autoscaling/v2beta1 to use HorizontalPodAutoscaler" }}
{{- end }}
```
Similar adaption is in: https://github.com/traefik/traefik-helm-chart/blob/859adf66447d8242ae6c697fc901473d4684c282/traefik/templates/ingressclass.yaml#L2-L10

This code should allow the template to adapt to older versions.